### DISCUSSION TITLE
Vec::normalize_or(fallback)

### DISCUSSION THREAD

**Discussion by TheComet** [bitshifter/glam-rs#483] [Ideas]
I ran into a case where it'd be convenient to be able to specify a fallback value for when normalize fails. I don't want acceleration or velocity to ever be zero, instead, I always want to get a unit vector:

```rust
      let mut random_unit_vector = || {
          Vec3::new(rng.gen_range(-1.0..1.0), 0., rng.gen_range(-1.0..1.0))
              .normalize_or(Vec3::X)
      };
      let velocity = random_unit_vector() * VELOCITY_SCALAR;
      let acceleration = random_unit_vector() * ACCELERATION_SCALAR;
```

I submitted a PR here: [https://github.com/bitshifter/glam-rs/pull/484](https://github.com/bitshifter/glam-rs/pull/484)

**Comment by bitshifter** [comment]
Merged #484
